### PR TITLE
Iteration 2

### PR DESCRIPTION
**Pull request by foleymichelle9**
Update StatTracker

### CODE REVIEW

File: lib/game_team.rb [hunk 1]
```diff
@@ -72,4 +72,10 @@ def percentage_visitor_wins
     ((visitor_wins.length / all_visitor_games.length.to_f) * 100).round(2)
   end
 
+  def percentage_ties
+  all_ties = @@all.find_all do |game_team|
```
Comment (kelshadarby) [inline lib/game_team.rb:76]: I think we can put these percentages into a module at some point.

File: lib/game.rb [hunk 2]
```diff
@@ -53,4 +53,44 @@ def lowest_total_score
...
+  def average_goals_by_season
+    seasons_and_goals_average = {}
+    find_all_seasons.each do |season|
+      sum_of_goals = find_games_by_season(season).sum do |game|
+        (game.away_goals + game.home_goals)
+      end
+      seasons_and_goals_average[season] = (sum_of_goals.to_f / count_of_games_by_season[season]).round(2)
+    end
+    seasons_and_goals_average
+  end
```
Comment (margoflewelling) [inline lib/game.rb:95]: Could probably refactor with reduce if we have time

File: test/fixtures/game_teams_truncated.csv [hunk 3]
```diff
@@ -4,3 +4,37 @@ game_id,team_id,HoA,result,settled_in,head_coach,goals,shots,tackles,pim,powerPl
...
+2012030314,5,away,LOSS,REG,Dan Bylsma,0,6,33,8,3,0,47.5,8,1
+2012030314,6,home,WIN,REG,Claude Julien,1,6,25,8,3,0,52.5,11,5
+2012030231,17,away,LOSS,REG,Mike Babcock,1,5,43,10,3,0,52.5,6,4
+2012030231,16,home,WIN,REG,Joel Quenneville,2,10,24,10,3,1,47.5,2,11
+2012030232,17,away,WIN,REG,Mike Babcock,2,7,26,6,4,0,55.4,6,6
+2012030232,16,home,LOSS,REG,Joel Quenneville,1,5,36,10,2,0,44.6,7,9
+2012030233,16,away,LOSS,REG,Joel Quenneville,1,10,22,22,4,0,43.9,5,6
+2012030233,17,home,WIN,REG,Mike Babcock,3,7,28,10,5,0,56.1,9,3
+2012030234,16,away,LOSS,REG,Joel Quenneville,0,7,22,10,3,0,48.3,8,4
+2012030234,17,home,WIN,REG,Mike Babcock,2,6,30,10,3,1,51.7,10,8
```
Comment (kelshadarby) [inline test/fixtures/game_teams_truncated.csv:27]: Nice! This is good data to test on.

File: test/game_team_test.rb [hunk 4]
```diff
@@ -4,7 +4,7 @@
 class GameTeamTest < Minitest::Test
 
   def setup
-    @game_teams = GameTeam.create_game_team('./test/fixtures/game_teams_truncated.csv')
+    @game_teams = GameTeam.create_game_teams('./test/fixtures/game_teams_truncated.csv')
```
Comment (kelshadarby) [inline test/game_team_test.rb:7]: Good consistency.

File: test/fixtures/games_truncated.csv [hunk 5]
```diff
@@ -4,3 +4,78 @@ game_id,season,type,date_time,away_team_id,home_team_id,away_goals,home_goals,ve
...
+2013020941,20132014,Regular Season,3/7/14,7,14,3,1,Audi Field,/api/v1/venues/null
+2012020320,20122013,Regular Season,3/5/13,1,10,2,2,WakeMed Soccer Park,/api/v1/venues/null
+2012020632,20122013,Regular Season,4/16/13,13,2,2,3,Centruy Link Field,/api/v1/venues/null
+2012020358,20122013,Regular Season,3/9/13,19,28,2,3,Banc of California Stadium,/api/v1/venues/null
+2013020887,20132014,Regular Season,2/28/14,10,2,2,3,Centruy Link Field,/api/v1/venues/null
+2013020924,20132014,Regular Season,3/5/14,13,6,1,2,Toyota Stadium,/api/v1/venues/null
+2012020205,20122013,Regular Season,2/17/13,9,10,0,3,WakeMed Soccer Park,/api/v1/venues/null
+2013021213,20132014,Regular Season,4/12/14,4,5,2,3,Children's Mercy Park,/api/v1/venues/null
+2013020972,20132014,Regular Season,3/11/14,52,21,2,3,BC Place,/api/v1/venues/null
+2013021119,20132014,Regular Season,3/30/14,20,9,3,4,Yankee Stadium,/api/v1/venues/null
```
Comment (margoflewelling) [inline test/fixtures/games_truncated.csv:81]: Added so we have both post and regular season games over multiple seasons

File: lib/stat_tracker.rb [hunk 6]
```diff
@@ -1,17 +1,14 @@
...
     game_teams_path = locations[:game_teams]
-
     StatTracker.new(game_path, team_path, game_teams_path)
   end
-
-  attr_reader :game_path, :team_path, :game_teams_path
+  attr_reader :games, :teams, :game_teams
   def initialize(game_path, team_path, game_teams_path)
-    @game_path = game_path
-    @team_path = team_path
```
Comment (kelshadarby) [inline lib/stat_tracker.rb:14]: Good for visibility.

File: test/team_test.rb [hunk 7]
```diff
@@ -22,5 +22,4 @@ def test_it_has_attributes
     assert_equal '/api/v1/teams/26', @team.link
 
   end
-
- end
+end
```
Comment (chrisstankus99) [inline test/team_test.rb:25]: Thanks for fixing this - it was triggering my ocd

File: lib/game.rb [hunk 8]
```diff
@@ -53,4 +53,44 @@ def lowest_total_score
   def biggest_blowout
     @@all.map { |game| (game.away_goals - game.home_goals).abs}.max
   end
+
+  def find_all_seasons
```
Comment (kelshadarby) [inline lib/game.rb:57]: Nice multi-use helper method.

**Pull request merged**